### DISCUSSION TITLE
Should we use pinia or useState?Pinia has a bug

### DISCUSSION THREAD

**Discussion by bbhxwl** [nuxt/nuxt#27724] [General]
Pinia has a bug
When refreshing the page, the request will not retrieve the pinia data in the cache, and it will only be mounted in pinia after loading is completed.

```
persist:true
```

**Comment by xjccc** [comment]
You should use `cookie` to store your data.

Because `localStorage` & `sessionStorage` is the `client api`.

So you can't use it in server.

**Reply by bbhxwl** [reply]
> You should use `cookie` to store your data.
> 
> Because `localStorage` & `sessionStorage` is the `client api`.
> 
> So you can't use it in server.


You misunderstood my meaning.
This setting is already the default cookie



```
export const useStore = defineStore('main', {
    state: () => {
        return {
        }
    },
    getters: {
    },
    actions: {
        
    },
    persist:true
});
```

**Reply by bbhxwl** [reply]
> You should use `cookie` to store your data.
> 
> Because `localStorage` & `sessionStorage` is the `client api`.
> 
> So you can't use it in server.

The persistence cycle seems to be slow

**Reply by xjccc** [reply]
Could you provide a mini repo for this?

**Reply by bbhxwl** [reply]
> Could you provide a mini repo for this?

https://github.com/nuxt/nuxt/issues/27765
